src/smt/theory_str_noodler/util.cc

```diff
@@ -556,21 +562,23 @@ namespace smt::noodler::util {
 
         if(is_variable(ex, m_util_s)) {
             std::string var = ex->get_decl()->get_name().str();
-            terms.emplace_back(BasicTermType::Variable, var);
+            BasicTerm bvar(BasicTermType::Variable, var);
+            terms.emplace_back(bvar);
+            var_name.insert({bvar, expr_ref(ex, m)});
```

Comment:
I would use `emplace` here to prevent unnecessary copies of the created structure.